sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/EncryptionCosmosAsyncContainer.java

```diff
@@ -2,73 +2,80 @@
         this.encryptor = encryptor;
     }
 
     private Mono<CosmosItemResponse<byte[]>> createItemStream(byte[] payload,
                                                               PartitionKey partitionKey,
-                                                              CosmosItemRequestOptions requestOptions) {
+                                                              EncryptionItemRequestOptions encryptionItemRequestOptions) {
         Preconditions.checkNotNull(payload, "payload can't be null");
 
         // TODO: add diagnostics
-        EncryptionItemRequestOptions encryptionItemRequestOptions = Utils.as(requestOptions, EncryptionItemRequestOptions.class);
-        if (encryptionItemRequestOptions != null && encryptionItemRequestOptions.getEncryptionOptions() != null) {
-
-            payload = EncryptionProcessor.encryptAsync(payload, encryptor, encryptionItemRequestOptions.getEncryptionOptions());
+        assert encryptionItemRequestOptions != null && encryptionItemRequestOptions.getEncryptionOptions() != null;
+        payload = EncryptionProcessor.encryptAsync(payload, encryptor, encryptionItemRequestOptions.getEncryptionOptions());
 
-            Mono<CosmosItemResponse<byte[]>> response = super.createItem(payload, partitionKey, requestOptions);
+        Mono<CosmosItemResponse<byte[]>> response = container.createItem(payload, partitionKey, encryptionItemRequestOptions);
 
-            return response
-                .subscribeOn(Schedulers.elastic())
-                .publishOn(Schedulers.elastic())
-                .map(rsp -> {
-                        EncryptionBridgeInternal.setByteArrayContent(rsp, decryptResponseAsync(EncryptionBridgeInternal.getByteArrayContent(rsp), encryptionItemRequestOptions.getDecryptionResultHandler()));
-                        return rsp;
-                    }
-                );
-
-        } else {
-            throw new NotImplementedException("TODO not implemented yet");
-            // TODO moderakh compelte the non encryption branch
-            // return super.createItem()
-        }
+        return response
+            .subscribeOn(Schedulers.elastic())
+            .publishOn(Schedulers.elastic())
+            .map(rsp -> {
+                    EncryptionModelBridgeInternal.setByteArrayContent(
+                        rsp,
+                        decryptResponseAsync(
+                            EncryptionModelBridgeInternal.getByteArrayContent(rsp),
+                            encryptionItemRequestOptions.getDecryptionResultHandler()));
+                    return rsp;
+                }
+            );
     }
 
     // TODO ensure all other apis call this guy
+
     /**
      * create item and encrypts the requested fields
-     * @param item the Cosmos item represented as a POJO or Cosmos item object.
-     * @param partitionKey the partition key.
+     *
+     * @param item           the Cosmos item represented as a POJO or Cosmos item object.
```

Comment:
This will probably fail the analyze build - because of multiple spaces between java docs. I can share my intellij code styling file - which you can import and when you refactor code through intellij, it should automatically solve these issues.

Reply:
thanks for pointing out. will fix and will ensure passes the analyzer.

Reply:
javadoc analyze passed now. addressed. thanks.